### PR TITLE
Replaced EventSubscriber by listener.

### PR DESCRIPTION
**Pull request by laurentmuller**
This is the change for issue #1388.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "require": {
         "php": "^8.1",
         "ext-simplexml": "*",
+        "doctrine/annotations": "^2.0",
```
Comment (garak) [inline composer.json:20]: I can't see any use of annotations... Do we really need to re-introduce this?

**Review by garak — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (laurentmuller) [reply composer.json:20]: This is used by the `VichUploaderExtension` class on function `registerAnnotationStrategy`. If not, a `MissingPackageException` is raised:

`Vich\UploaderBundle\Exception\MissingPackageException : Annotations support missing. Try running "composer require doctrine/annotations".`

Comment (garak) [reply composer.json:20]: That's a conditional requirement, which is already covered in our "suggest" section. Please remove it from "require"

Comment (laurentmuller) [reply composer.json]: OK.

**Comment by garak** [comment]
I guess the test failure depends on some lower dependencies that need to be raised

File: src/Adapter/ORM/DoctrineORMAdapter.php [hunk 2]
```diff
@@ -22,7 +22,7 @@ public function recomputeChangeSet(BaseLifecycleEventArgs $event): void
     {
         $object = $event->getObject();
 
-        $em = $event->getEntityManager();
+        $em = $event->getObjectManager();
```
Comment (garak) [inline src/Adapter/ORM/DoctrineORMAdapter.php:25]: This change is about a different issue, it should be in another PR

Comment (laurentmuller) [reply src/Adapter/ORM/DoctrineORMAdapter.php:25]: You are right. How we can move it to the #1393 PR?

**Review by garak — APPROVED** [review]

File: src/DependencyInjection/VichUploaderExtension.php [hunk 3]
```diff
@@ -196,7 +196,7 @@ protected function registerListeners(ContainerBuilder $container, array $config)
     {
         $servicesMap = [
             'inject_on_load' => ['name' => 'inject', 'priority' => 0, 'events' => ['postLoad']],
-            'delete_on_update' => ['name' => 'clean', 'priority' => 50, 'events' => ['preRemove', 'postFlush']],
+            'delete_on_update' => ['name' => 'clean', 'priority' => 50, 'events' => ['preUpdate']],
```
Comment (garak) [inline src/DependencyInjection/VichUploaderExtension.php:199]: why?

**Review by garak — CHANGES_REQUESTED** [review]

**Comment by Tofandel** [comment]
Can we add a test to the extension to check that the listeners are registered?

**Review by garak — APPROVED** [review]
